### PR TITLE
Adds Bulk API notes.

### PR DESCRIPTION
**Pull request by renatoliveira**
I thought of adding these Bulk API notes to help anyone who might be struggling a bit with the `sf.bulk.` methods of the library. In this text I write about a common error ("InvalidBatch") and how to check the data before sending it over to Salesforce, and also add some tips on preparing the data's date/time columns and relationship fields.

### CODE REVIEW

File: docs/user_guide/bulk_api_notes.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
+====================
+Bulk API Usage Notes
+====================
+
+It is worth noting that the Bulk API is a common resource to build applications that extract or import data regularly on the Salesforce Platform. Its error messages can sometimes be cryptic, however. For example, if you try to insert a record with a duplicate external ID, you will get an error message that says "Duplicate External ID specified: [External ID Field Name]". This is not very helpful, as it does not tell you which record is causing the problem. You will need to query the records to find the duplicate. This is a common issue with the Bulk API, and it is worth noting that you will need to handle these errors in your application.
+
+Exception Code "InvalidBatch" with message "Records not processed"
+==================================================================
+
+When processing your Bulk API request, one of the first chings Salesforce does is to check the schema of your payload. If the schema is not correct, you will get an error message with the exception code "InvalidBatch" and the message "Records not processed".
```
Comment (mattmartin26) [inline docs/user_guide/bulk_api_notes.rst:10]: Misspelling: "chings"

File: docs/user_guide/bulk_api_notes.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+If you are upserting data that looks up to existing data in Salesforce through an external id, make sure you define the dictionary object in the Python code before calling the *sf.bulk.* methods. For example, if you have a custom object called "MyCustomObject__c" that has a lookup relationship to the Account object, and you are using the `AccountNumber` field as the external id, you will need to define the dictionary object like this:
+
+.. code-block:: python
+
+    my_dataframe['Account__r'] = my_dataframe['Account__r'].apply(lambda x: {'AccountNumber': x})
+
+This will convert the `Account__r` column to a dictionary with the external id field that you are using to look up the Account object. This is because of the way relationships work in Salesforce. If you are somewhat familiar with how the platform's database works, you will notice that for the row that is going to be inserted the column that references the lookup object is transformed to an object, in a similar way that in Apex you use the relationship field to access the fields of the related object. This works for standard objects as well, where instead of *ContactId* you would use just *Contact*.
```
Comment (mattmartin26) [inline docs/user_guide/bulk_api_notes.rst:38]: Finding this sentence just a bit unclear. Unsure how to suggest an improvement but for someone unexperienced in Apex its hard to envision what being 'transformed to an object' means.

**Review by mattmartin26 — APPROVED** [review]
Overally, VERY needed addition to the documentation!! Please approve this PR.

File: docs/user_guide/bulk_api_notes.rst (hunk 2 above)
Comment (renatoliveira) [reply docs/user_guide/bulk_api_notes.rst]: I made some changes to this PR so that the spelling is fixed and this part is a little better explained.

**Pull request merged**